atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/knowledge/KnownConcludedTransactionsStore.java

```diff
@@ -117,6 +121,27 @@ public void supplement(Set<Range<Long>> timestampRangesToAdd) {
         throw new SafeIllegalStateException("Unable to supplement set of concluded timestamps.");
     }
 
+    /**
+     * Modifies a range that contains the minimum timestamp to have it's lower-bound set (exclusive) to the minimum timestamp.
+     *
+     * This is needed as we do not want to not conclude a range that encompasses the minimum timestamp.
```

Comment:
We should probably add a comment _somewhere_ explaining the minimum timestamp in more detail.

Reply:
My vote would be in `TimestampRangeSet`